File: backend/app/models/briefing.py
from typing import Optional
from pydantic import BaseModel

from app.models.context import InputTask


class BriefingRequest(BaseModel):
    date: Optional[str] = None
    tasks: list[InputTask]
    max_learn: int = 5
    max_pulse: int = 5
    max_tools: int = 3


class SnippetItem(BaseModel):
    title: str
    url: str
    source: str
    relevance_note: str


class DailySnippet(BaseModel):
    date: str
    topics: list[str]
    learn: list[SnippetItem]
    pulse: list[SnippetItem]
    tools: list[SnippetItem]

File: mobile/src/api/briefing.ts
import type { InputTask } from './context';

import { API_BASE as BASE_URL } from '../config';
const API_BASE = `${BASE_URL}/api`;

export type SnippetItem = {
  title: string;
  url: string;
  source: string;
  relevance_note: string;
};

export type DailySnippet = {
  date: string;
  topics: string[];
  learn: SnippetItem[];
  pulse: SnippetItem[];
  tools: SnippetItem[];
};

export async function fetchDailySnippet(tasks: InputTask[]): Promise<DailySnippet> {
  const controller = new AbortController();
  const timeout = setTimeout(() => controller.abort(), 40000); // 40s for cold start

  console.log('Fetching daily snippet with tasks:', tasks);

  try {
    const res = await fetch(`${API_BASE}/briefing/daily`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ tasks, max_learn: 5, max_pulse: 5, max_tools: 3 }),
      signal: controller.signal,
    });

    if (!res.ok) {
      throw new Error(`Briefing request failed: ${res.status}`);
    }

    return res.json();
  } finally {
    clearTimeout(timeout);
  }
}
